src/server/network.ts

```diff
@@ -275,15 +288,19 @@ export class Response extends SdkObject {
   private _headersMap = new Map<string, string>();
   private _getResponseBodyCallback: GetResponseBodyCallback;
   private _timing: ResourceTiming;
+  private _serverIPAddressAndPort: IPAddressAndPort;
+  private _securityDetails?: SecurityDetails;
```

Comment:
@JoelEinbinder Should we combine all this info? Or leave it as separate fields? Also, what's the consensus on using  making all members of SecurityDetails/IPAddressAndPort optional instead of using the `-1/''` special values.

Reply:
id prefer optional instead of special values

Reply:
Done! I got to delete a lot of awkward code :)